### PR TITLE
Task/vendor balance table

### PR DESCRIPTION
**Pull request by smyaseen**
## GOAL && WORK

Implement complete balance APIs on Vendor

### CODE REVIEW

File: src/Features/Vendor/BalanceSheet/EditBalance/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      },
+    },
+  ]);
+
+  const saveHandler = () => {
+    const { validateArray, isValid } = validateOnSubmit(fields, true);
+    setFields(validateArray);
+    const [{ value: amount }] = fields;
+
+    console.log(amount);
```
Comment (SiddiquiMonis401) [inline src/Features/Vendor/BalanceSheet/EditBalance/index.js:32]: Remove console and comments

File: src/Features/Vendor/BalanceSheet/requests.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+
+export const FetchBalances = () => {
+  const vendorId = useSelector((state) => state.authReducer.id);
+  return useQuery('balance', async () => {
+    const {
+      data: { results },
+    } = await axios.get('http://localhost:4000/v1/balance');
```
Comment (SiddiquiMonis401) [inline src/Features/Vendor/BalanceSheet/requests.js:10]: Can't we use the Base_URL constants here ?

File: src/Features/Vendor/BalanceSheet/requests.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    return results.filter((res) => res.vendorId === vendorId);
+  });
+};
+
+export const FetchBalanceById = (id) => {
+  const vendorId = useSelector((state) => state.authReducer.id);
+  return useQuery('balance', async () => {
+    const {
+      data: { results },
+    } = await axios.get('http://localhost:4000/v1/balance/' + id);
```
Comment (SiddiquiMonis401) [inline src/Features/Vendor/BalanceSheet/requests.js:20]: if Id is there it should be included in the key as well here !!

File: src/routes/RouteNames.js [hunk 4]
```diff
@@ -28,6 +28,8 @@ const RouteNames = {
   addmenu: '/addmenu',
   editmenu: '/editmenu',
   orderHistories: '/orderHistories',
+  balanceSheet: '/balanceSheet',
+  editBalanceSheet: '/editBalanceSheet',
```
Comment (SiddiquiMonis401) [inline src/routes/RouteNames.js:32]: please try to make sure that all routes in the project are hyphenated like edit-balance-sheet

**Review by SiddiquiMonis401 — CHANGES_REQUESTED** [review]

**Review by ShahirZain — APPROVED** [review]

**Review by SiddiquiMonis401 — APPROVED** [review]

**Pull request merged**